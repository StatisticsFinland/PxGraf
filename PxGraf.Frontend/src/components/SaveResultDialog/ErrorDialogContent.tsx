import { useTranslation } from 'react-i18next';

import {
    Alert, DialogContent,
  } from '@mui/material';
import React from 'react';

export const ErrorDialogContent: React.FC = () => {
    const { t } = useTranslation();

    return (
        <DialogContent dividers={true}>
            <Alert severity="error"> {t('saveResultDialog.errorMessage')} </Alert>
        </DialogContent>
    );
}

export default ErrorDialogContent;