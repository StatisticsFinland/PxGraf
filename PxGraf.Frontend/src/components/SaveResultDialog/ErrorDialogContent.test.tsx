import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ErrorDialogContent } from './ErrorDialogContent';

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => null),
            },
        };
    },
}));

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<ErrorDialogContent />);
        expect(asFragment()).toMatchSnapshot();
    });
});