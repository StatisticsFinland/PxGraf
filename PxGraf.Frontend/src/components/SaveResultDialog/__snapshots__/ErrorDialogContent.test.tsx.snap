// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiDialogContent-root MuiDialogContent-dividers css-1t4vnk2-MuiDialogContent-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-colorError MuiAlert-standardError MuiAlert-standard css-x7o5da-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ErrorOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
         saveResultDialog.errorMessage 
      </div>
    </div>
  </div>
</DocumentFragment>
`;
